bindings/pyroot_experimental/pythonizations/python/ROOT/_facade.py

```diff
@@ -175,6 +180,32 @@ def _setattr(self, name, val):
 
         return setattr(self, name, val)
 
+    def _run_rootlogon(self):
+        hasargv = hasattr(sys, 'argv')
+        # custom logon file (must be after creation of ROOT globals)
+        if hasargv and not '-n' in sys.argv and not self.PyConfig.DisableRootLogon:
```

Comment:
This `if` relies on the fact (or implementation detail?) that Python evaluates the `if` cases from left to right. Otherwise, the `not '-n' in sys.argv` will fail with an `AttributeError`. Just to point out ;) I'm not 100% sure whether this can be regarded as safe.

Reply:
As per the Python documentation:
https://docs.python.org/3/reference/expressions.html#boolean-operations

```
The expression x and y first evaluates x; if x is false, its value is returned;
otherwise, y is evaluated and the resulting value is returned.
```

Reply:
Nice! Didn't know about that!